### PR TITLE
[SPARK-44316][BUILD] Upgrade Jersey to 2.40

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade Jersey from 2.36 to 2.40.

### Why are the changes needed?
1.This version adapts to ASM9.5, which is also used by Spark currently
[Adopt ASM 9.5](https://github.com/eclipse-ee4j/jersey/pull/5305)

2.Also fix some bugs, eg:
[Fix possible NPE in netty client](https://github.com/eclipse-ee4j/jersey/pull/5330)
[Get media type fix](https://github.com/eclipse-ee4j/jersey/pull/5282)

3.Security vulnerability fix:
[CVE for dependency jackson-databind](https://github.com/eclipse-ee4j/jersey/issues/5225)

4.Full Release Notes:
https://github.com/eclipse-ee4j/jersey/releases/tag/2.40
https://github.com/eclipse-ee4j/jersey/releases/tag/2.39
https://github.com/eclipse-ee4j/jersey/releases/tag/2.38
https://github.com/eclipse-ee4j/jersey/releases/tag/2.37

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -196,7 +196,11 @@
     <datanucleus-core.version>4.1.17</datanucleus-core.version>
     <guava.version>14.0.1</guava.version>
     <janino.version>3.1.9</janino.version>
-    <jersey.version>2.36</jersey.version>
+    <!--
```
Comment (LuciferYang) [inline pom.xml:199]: jersey 3.x only support Java 11+, maybe we can upgrade to 3.x in Spark 4.0 :)

**Review by LuciferYang — APPROVED** [review]
LGTM

File: pom.xml (hunk 1 above)
Comment (panbingkun) [reply pom.xml:199]: Good proposal!

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @panbingkun and all.

File: pom.xml (hunk 1 above)
Comment (dongjoon-hyun) [reply pom.xml:199]: +1 for that.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.5.0.

**Pull request closed without merging**